src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,347 @@
+import React, {
+  createElement,
+  ForwardRefExoticComponent,
+  ReactElement,
+  ReactNode,
+  RefObject,
+  useEffect,
+  useRef,
+  useState,
+} from 'react'
+import classnames from 'classnames'
+import { isElementInViewport } from './utils'
+
+type TooltipProps<T> = {
+  label: string
+  position?: 'top' | 'bottom' | 'left' | 'right' | undefined
+  wrapperclasses?: string
```

Comment:
i removed className prop, as i didn't see the purpose of having both className and wrapperclasses. may have lost some perspective  for why we'd need a separate prop since starting this ticket so your input would be appreciated @suzubara

Reply:
The `className` prop is meant to allow a user to pass a custom CSS class to the trigger element, so that value should be incorporated in the list of classes passed to the button or custom element used. `wrapperClasses` is the same, but for the wrapper element instead (this mirrors the `data-classes` attribute exposed by USWDS).

For example if a user of USWDS did this:
```
    <button type="button" class="usa-button usa-tooltip myCustomButtonClass" data-position="top" data-classes="width-full tablet:width-auto" title="Top">Show on top</button>
```

they should be able to achieve the same result using our Tooltip component with this:
```
    <Tooltip className="myCustomButtonClass" position="top" wrapperClasses="width-full tablet:width-auto" label="Top">Show on top</button>
```

and that should result in the `tooltipWrapper` element having the wrapperClasses value (`width-full tablet:width-auto`), and the `triggerElement` element having the className value (`myCustomButtonClass`) in addition to the other usa tooltip classes

Reply:
awesome, I think my latest change should fix it @suzubara